src/Auth/SaveHandlers/Redis.php

```diff
@@ -80,7 +80,8 @@ public function close()
      */
     public function read($identifier)
     {
-        return $this->redis->get($this->getSessionKey($identifier));
+        $result = $this->redis->get($this->getSessionKey($identifier));
+        return $result === false ? 'empty' : $result;
```

Comment:
Yes, it must be string, but why it contains the 'empty' word?
http://php.net/manual/en/sessionhandlerinterface.read.php
```
Returns an encoded string of the read data. If nothing was read, it must return an empty string. Note this value is returned internally to PHP for processing.
```

Maybe just?:
```
return (string) $this->redis->get($this->getSessionKey($identifier));
```